### ISSUE TITLE
Localstack seems not to set policy attributes to SQS queue

### ISSUE THREAD

**Issue by djingoule** [localstack/localstack#804]
Hello,

I am trying to set a policy attribute to a SQS queue I have created. I don't get any error with these commands:
```
$ sqs_policy='{
    "Version":"2012-10-17",
    "Statement":[
      {
        "Effect":"Allow",
        "Principal": { "AWS": "*" },
        "Action":"sqs:SendMessage",
        "Resource":"'$sqs_queue_arn'",
        "Condition":{
          "ArnEquals":{
            "aws:SourceArn":"'$sns_topic_arn'"
          }
        }
      }
    ]
  }'
$ sqs_policy_escaped=$(echo $sqs_policy | perl -pe 's/"/\\"/g')
$ sqs_attributes='{"Policy":"'$sqs_policy_escaped'"}'
$ awslocal sqs set-queue-attributes \
  --queue-url "$sqs_queue_url" \
  --attributes "$sqs_attributes"
```
But when I check the attribute of the queue with these commands, it seems like there is no policy attribute
```
$ awslocal sqs get-queue-attributes --queue-url $sqs_queue_url --attribute-names Policy
$ awslocal sqs get-queue-attributes --queue-url $sqs_queue_url --attribute-names All
{
    "Attributes": {
        "ApproximateNumberOfMessagesNotVisible": "1", 
        "ApproximateNumberOfMessagesDelayed": "0", 
        "CreatedTimestamp": "1528886541", 
        "ApproximateNumberOfMessages": "0", 
        "ReceiveMessageWaitTimeSeconds": "20", 
        "DelaySeconds": "0", 
        "VisibilityTimeout": "300", 
        "LastModifiedTimestamp": "1528886541", 
        "QueueArn": "arn:aws:sqs:elasticmq:000000000000:TestQueue"
    }
}
```
However, when I subscribe the queue to a SNS topic, and I publish a message to the topic, that message is sent to the SQS queue.
I am using 0.8.6.2 version of localstack and run it with docker. 

At the end, what I am trying to do is to publish an event from S3 to SQS passing by SNS. 

I don't know if I am doing something wrong of if it a localstack bug/limitation.

**Comment by dchw** [comment]
Running into a similar issue here, but doing it via code. In C#:

```
public async Task LinkToBucket(string bucketName) =>
			await _client.AuthorizeS3ToSendMessageAsync (_queueUrl, bucketName);
```

This doesn't add any policy, [but the code from the AWS SDK suggests it should.](https://github.com/aws/aws-sdk-net/blob/b691e46e57a3e24477e6a5fa2e849da44db7002f/sdk/src/Services/SQS/Custom/_async/AmazonSQSClient.Extension.cs#L57). Would love for this issue to be fixed... also wish I had the time to dig in to try and fix it.

**Comment by timetorock** [comment]
@djingoule Do you have any updates on this? Same for me via go-aws-sdk

**Comment by djingoule** [comment]
> @djingoule Do you have any updates on this? Same for me via go-aws-sdk

No @timetorock ,no updates for now. Would really love for this issue to be fixed.

**Comment by whummer** [comment]
Thanks for reporting @djingoule @timetorock . This should be fixed in #1520 - please reopen this issue if the problem persists. Thanks

**Comment by wederribas** [comment]
It seems that a regression was introduced that allowed this issue to occur again.

I have an SNS <-> SQS setup [here](https://github.com/weder-ribas-ms-thesis/cs7990-hummingbird-app/blob/865082ec0bff6efc1c7dd9caadcb4bebdab5a044/terraform/modules/eventing/main.tf#L37). The SQS queue policy is attached correctly, and it uses the following structure:

```tf
data "aws_iam_policy_document" "sns_topic_policy" {
  statement {
    sid     = "SNSPublishToSQS"
    actions = ["sqs:SendMessage"]
    effect  = "Allow"

    principals {
      type        = "AWS"
      identifiers = ["*"]
    }

    resources = [
      aws_sqs_queue.media_management_sqs_queue.arn
    ]

    condition {
      test     = "ArnEquals"
      variable = "aws:SourceArn"
      values   = [aws_sns_topic.media_management_topic.arn]
    }
  }
}
```

When I fetch the SQS queue policy attributes, I do get the policy back:
```sh
awslocal sqs get-queue-attributes --queue-url http://sqs.us-west-2.localhost.localstack.cloud:4566/000000000000/media-management-sqs-queue --attribute-names Policy --query 'Attributes.Policy' --output text | jq .
```

```json
{
  "Statement": [
    {
      "Action": "sqs:SendMessage",
      "Condition": {
        "ArnEquals": {
          "aws:SourceArn": "arn:aws:sns:us-west-2:000000000000:media-management-topic"
        }
      },
      "Effect": "Allow",
      "Principal": {
        "AWS": "*"
      },
      "Resource": "arn:aws:sqs:us-west-2:000000000000:media-management-sqs-queue",
      "Sid": "SNSPublishToSQS"
    }
  ],
  "Version": "2012-10-17"
}
```

However, LocalStack raises a request denied from SNS to publish a message to SQS.

Here's the error from EventStudio:
```
Request for service 'SQS' for operation 'SendMessage' denied.

"Action": 'sqs:SendMessage' for 'arn:aws:sqs:us-west-2:000000000000:media-management-sqs-queue'

Necessary permissions for this action:
Explicitly allowed permissions: 0
Explicitly denied permissions: 0
Implicitly denied permissions: 1
```

It works fine on AWS.
